python/lsst/ts/ess/csc/ess_csc.py

```diff
@@ -175,12 +179,8 @@ async def process_temperature_telemetry(
         for i, value in enumerate(sensor_data):
             # The telemetry channels start counting at 1 and not 0.
             telemetry[f"temperatureC{i + 1:02d}"] = value
-        self.log.info(f"Received temperatures {telemetry}")
-        self.log.info("Sending telemetry.")
-        telemetry_method = getattr(
-            self, f"tel_temperature{device_configuration.num_channels}Ch"
-        )
-        telemetry_method.set_put(**telemetry)
+        self.log.debug(f"Sending telemetry {telemetry}")
```

Comment:
I think the code `telemetry[f"temperatureC{i + 1:02d}"] = value` (just above this line) needs to be changed to set the key "temperatures" to an array of temperatures. Something along the lines of:
```
temperatures = list(NAN_TEMPERATURES)
temperatures[:num_channels] = array-of-num_channels-values
telemetry["temperature"] = temperatures
telemetry["numChannels"] = num_channels
```
Where you set NAN_TEMPERATURES = (math.nan,)*16 somewhere else.

In theory you could ignore the extra channels, since readers should not read data past numChannels values. But it's a safer to be explicit.

Reply:
I just pushed a code change that addresses exactly that.